Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift

```diff
@@ -553,3 +515,82 @@ extension DocumentationContentRenderer {
     }
 
 }
+
+private extension Array where Element == DeclarationRenderSection.Token {
+    // Replaces kind "typeIdentifier" with "identifier" if the fragments matches the pattern:
+    // [keyword=class,protocol,enum,typealias,etc.][ ]([typeIdentifier=x_i)][.])*[typeIdentifier=x_i],
+    // where the x_i joined with separator "." equal the `symbolTitle`
+    func mapNameFragmentsToIdentifierKind(matching symbolTitle: String) -> Self {
+        let (includesTypeOrExtensionDeclaration, nameRange) = self.typeOrExtensionDeclaration()
+        
+        if includesTypeOrExtensionDeclaration
+            && self[nameRange].map(\.text).joined() == symbolTitle {
+            return self.enumerated().map { (index, token) -> DeclarationRenderSection.Token in
+                
+                if nameRange.contains(index) && token.kind == .typeIdentifier {
+                    return DeclarationRenderSection.Token(
+                        text: token.text,
+                        kind: .identifier,
+                        preciseIdentifier: token.preciseIdentifier
+                    )
+                }
+                
+                return token
+            }
+        }
+        
+        return self
+    }
+}
+
+private extension Collection where Element == DeclarationRenderSection.Token, Index == Int {
+    func typeOrExtensionDeclaration() -> (includesTypeOrExtensionDeclaration: Bool, name: Range<Index>) {
+        self.reduce(into: TypeOrExtensionDeclarationNameExtractionSM(), { sm, token in sm.consume(token) }).result()
+    }
+}
+
+private enum TypeOrExtensionDeclarationNameExtractionSM {
```

Comment:
My professor wouldn't be happy to hear that 😅 ...resolved in 6f7062dd14eb90c74b64d9e15d2907f48dc9fdc3.

Reply:
Haha 😄

The flow was mostly linear—with one small loop and early returns for error handling—but I kept reading the state transitions a couple of times to see if I was missing any transitions. 
```
┌─────────────┐  ┌─────────────┐  ┌─────────────┐  ┌─────────────┐  ┌─────────────┐   
│   initial   │─▶│found keyword│─▶│  expect ID  │─▶│  found ID   │─▶│    done     │──┐
└─────────────┘  └─────────────┘  └─────────────┘  └─────────────┘  └─────────────┘◀─┘
       │                │                │   ▲        │   │                           
       │                │                │   └────────┘   │         ┌─────────────┐   
       └────────────────┴────────────────┴────────────────┴────────▶│    error    │──┐
                                                                    └─────────────┘◀─┘
```
With the procedural code I felt that I could skim that structure more easily but that could very well be because it's been ~10 years since I last wrote a state machine 😉